File: backend/java-employee-creator/src/main/java/com/lgsoftdev/javaemployeecreator/dto/EmployeeStatusDto.java
package com.lgsoftdev.javaemployeecreator.dto;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.Date;

@Data
public class EmployeeStatusDto {
    private Integer id;

    private Integer employeeId;

    @NotNull
    private Integer contractType;

    @NotNull
    private Date startDate;

    private Date finishDate;

    @NotNull
    private Integer workType;

    private Short hoursPerWeek;
}

File: frontend/react-employee-creator/src/dtos/EmployeeStatusDto.ts
class EmployeeStatusDto {
  id?: number;
  employeeId?: number;
  contractType?: number;
  startDate: Date;
  finishDate?: Date;
  workType?: number;
  hoursPerWeek?: number;

  constructor(
    id: number | undefined,
    contractType: number | undefined,
    startDate: Date,
    workType?: number | undefined,
    employeeId?: number | undefined,
    finishDate?: Date | undefined,
    hoursPerWeek?: number | undefined
  ) {
    this.id = id;
    this.employeeId = employeeId;
    this.contractType = contractType;
    this.startDate = startDate;
    this.finishDate = finishDate;
    this.workType = workType;
    this.hoursPerWeek = hoursPerWeek;
  }
}

export default EmployeeStatusDto;
